rotkehlchen/api/rest.py

```diff
@@ -4234,3 +4234,41 @@ def get_ens_avatar(self, ens_name: str, match_header: Optional[str]) -> Response
                 )
 
         return maybe_create_image_response(image_path=avatar_path)
+
+    def clear_cache(
+            self,
+            cache_type: Literal['icons', 'avatars'],
+            entries: Optional[list[str]],
+    ) -> Response:
+        """Removes cached items from the filesystem."""
+        if cache_type == 'icons':
+            cache_dir = self.rotkehlchen.icon_manager.icons_dir
+        else:  # can only be avatars
+            cache_dir = self.rotkehlchen.icon_manager.icons_dir / 'avatars'
+
+        # delete every file in the cache directory.
+        if entries is None or len(entries) == 0:
+            for entry in cache_dir.iterdir():
+                # don't delete avatars dir when clearing icons cache
+                if (
+                    cache_type == 'icons' and
+                    entry.parent == self.rotkehlchen.icon_manager.icons_dir / 'avatars'  # noqa: E501
+                ):
+                    continue
+
+                if entry.is_file():
+                    entry.unlink()
+
+            return api_response(OK_RESULT)
+
+        if cache_type == 'icons':
+            for identifier in entries:
+                self.rotkehlchen.icon_manager.delete_icon(Asset(identifier))
+            return api_response(OK_RESULT)
```

Comment:
Question. Shouldn't an error response be returned if no icon/avatar was found for the requested identifier?

Reply:
I don't think that should be the case since the frontend doesn't know if an avatar/icon exists in the first instance

Reply:
Yes, probably makes sense. But pinging @rotki/frontend just to make sure.

Reply:
For these stuff ask in discord. Seeing a ping here may take time. I also think it's fine if there is nothing special returned in case missing. Result is the same. No file for this identifier.

Reply:
We don't need any error code if the icon was missing in the first place. (Anything other than 200).

The only case IMO that would make sense to get an error is if the deletion fails, just to notify the user that the action was not successful.